gsmmodem/pdu.py

```diff
@@ -564,7 +558,10 @@ def _decodeAddressField(byteIter, smscField=False, log=False):
             if smscField:
                 addressValue = decodeSemiOctets(byteIter, addressLen-1)
             else:
-                addressLen = nibble2octet(addressLen)
+                if addressLen % 2:
+                    addressLen = int(addressLen / 2) + 1
+                else:
+                    addressLen = int(addressLen / 2)
```

Comment:
Just a question: Shouldn't `addressLen = int((addressLen + 1) / 2)` give the same values, just without the need of an `if` statement? (See https://github.com/babca/python-gsmmodem/issues/33#issuecomment-284520738)